ngx_http_sxg_filter_module.c

```diff
@@ -303,38 +333,123 @@ static ngx_str_t extract_angled_url(char* str, size_t len) {
   return (ngx_str_t){.data = NULL, .len = 0};
 }
 
-static void extract_preaload_url_list(ngx_str_t* link, ngx_array_t* const dst,
-                                      ngx_http_request_t* r) {
+// TODO(kumagi): Complex logic should be migrated to ngx_sxg_utils.c.
```

Comment:
This logic is complex -- can you unit test this?

Reply:
Yes, but I want separate it to refactoring work (in other patch).